base/bucket_gocb.go

```diff
@@ -1101,12 +1101,22 @@ func (bucket CouchbaseBucketGoCB) UpdateXattr(k string, xattrKey string, exp uin
 	// This is the only use case for macro expansion today - if more cases turn up, should change the sg-bucket API to handle this more generically.
 	xattrCasProperty := fmt.Sprintf("%s.cas", xattrKey)
 	worker := func() (shouldRetry bool, err error, value interface{}) {
-		// If the body doesn't exist, we need to set the AccessDelete flag to mutate the xattr.  If the body exists (and we're trying to delete it), revert
-		// to SubdocDocFlagNone
-		mutateFlag := gocb.SubdocDocFlagAccessDeleted
+
+		var mutateFlag gocb.SubdocDocFlag
 		if deleteBody {
+			// If the body exists (and we're trying to delete it), use SubdocDocFlagNone
 			mutateFlag = gocb.SubdocDocFlagNone
+		} else {
+			if cas == 0 {
+				// If the cas is 0, that means the document doesn't exist, in which case we are trying to set
+				// Xattrs on a non-existent document, so we must set the SubdocDocFlagMkDoc flag
+				mutateFlag = gocb.SubdocDocFlagMkDoc
+			} else {
+				// If the body doesn't exist, we need to set the AccessDelete flag to mutate the xattr.
```

Comment:
More correct to say "If the body *may* not exist, we need to set SubdocDocFlagAccessDeleted"